lookbackoptions_validation

```diff
@@ -0,0 +1,163 @@
+        events.append(
+            {
+                "track": "",
+                "time": fixing_time,
+                "op": None,
+                "quantity": 0,
+                "unit": "UPDATE",  
+            }
+        )
+    
+    events.append(
+        {
+            "track": "",
+            "time": maturity,
+            "op": None,
+            "quantity": 0,
+            "unit": "FIX",  
+        }
+    )
+    
+    events.append(
+        {
+            "track": "",
+            "time": maturity,
+            "op": "+",
+            "quantity": 1,
+            "unit": "LOOKBACK", 
+        }
+    )
+    # Defining look-back put payoff function
+
+    def lookback_put_pay_fn(inputs):
+        [s_min, k] = inputs
+        return [np.maximum(k - s_min, 0)]
+
+    # Defining the strike phrase, which is a fixed strike price
+    def strike_fn(inputs):
+        return [strike]
+
+    events_table = pa.RecordBatch.from_pylist(events, schema=TS_EVENT_SCHEMA)
+    return {
+        "events": events_table,
+        "expressions": {
+            "LOOKBACK": {
+                "type": "phrase",
+                "inp": ["MIN_PRICE", "K"],
+                "fn": lookback_put_pay_fn,
+            },
+            "UPDATE": {
+                "type": "snapper",
+                "inp": [ticker, "MIN_PRICE"],
+                "fn": lambda inputs: [np.minimum(inputs[0], inputs[1])],
+                "out": ["MIN_PRICE"],
+            },
+            "INIT": {
+                "type": "snapper",
+                "inp": [ticker],
+                "fn": lambda inputs: [inputs[0]],
+                "out": ["MIN_PRICE"],
+            },
+            "FIX": {
+                "type": "snapper",
+                "inp": [ticker],
+                "fn": strike_fn,
+                "out": ["K"],
+            },
+        },
+    }
+
+MC_PARAMS = {
+    "PATHS": 100_000,
+    "TIMESTEP": 1 / 250,
+    "SEED": 1,
+}
+
+# Provided data functions
+def basic_info():
+
+    return {
+        "prc_dt": datetime(2005, 9, 14),
+        "ticker": "SPX",
+        "ccy": "USD",
+        "spot": 100,  # Using spot price as 100
+    }
+
+def assets_data(rate=0.1):
+    info = basic_info()
+
+    div_rate = 0.02
+    times = np.array([0.0, 5.0])
+    rates = np.array([rate, rate])
+    discount_data = ("ZERO_RATES", np.column_stack((times, rates)))
+
+    fwds = info["spot"] * np.exp((rates - div_rate) * times)
+    fwd_data = ("FORWARDS", np.column_stack((times, fwds)))
+
+    return {info["ccy"]: discount_data, info["ticker"]: fwd_data}
+
+def localvol_data(rate=0.1, vol_file="spx_svi_2005_09_15.csv"):
+    info = basic_info()
+
+    svidf = pd.read_csv(vol_file)
+
+    return {
+        "BASE": "USD",
+        "PRICING_TS": py_to_ts(info["prc_dt"]).value,
+        "ASSETS": assets_data(rate=rate),
+        "MC": MC_PARAMS,
+        "LV": {"ASSET": "SPX", "VOL": svidf},
+    }
+
+# Validation of Local Vol for parameters S0=100, r=0.1, T=0.2 Yrs, m=4
+ticker = "SPX"
```

Comment:
move this file to src/contracts.
Eventually you'll have multiple files:

├── src
│   └── contracts                  # new folder
│       ├── lookback.py            # define lookback_put_timetable here
│       └── lookback_validate.py   # define validation dataset and code here, import loookback_put_timetable
├── 04_LookbackOptions.ipynb        # Notebook, use data from /data like the other notebooks, import loookback_put_timetable